### PR TITLE
Search: Add MockUsageMeter along with DonutMeterContainer component

### LINKED ISSUE #26052: JS Components: Build DonutMeter container
A container for the [DonutMeter component](https://github.com/Automattic/jetpack/issues/25917) that adds the necessary labels, links, and layout handling.

Image

**Dynamic Properties:**

- localized title string
- callback for “info” button
- callback for “Show details” button

Please ensure the component can be previewed in Storybook.

### PR DESCRIPTION
**Pull request by a8ck3n**
Fixes #26052

#### Changes proposed in this Pull Request:

* Introduce `DonutMeterContainer` component.
* Introduce `MockUsageMeter` to Jetpack Search Dashboard.
* Use pricing gate `isNewPricing202208` to display the `MockUsageMeter` component.

> Note by @dognose24: Since this DonutMeterContainer component may stay inside the Search package, I would prefer to complete the whole MockUsageMeter along with the DonutMeterContainer component to avoid redundant PR processes.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

Per discussion in Slack, we're going to build this directly in the Search package for now and look at Storybook integration as a follow up. It looks like we can build components as part of our package and still expose them to Storybook.

#### Does this pull request change what data or activity we track or use?

N/A

#### Testing instructions:

* Navigate the site URL with `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`.
* Ensure there will be a `UsageMeter` section displayed.
* Ensure the `UsageMeter` will be in line with the design.

| Before | After |
| --- | ----------- |
| 截圖 2022-10-03 上午9 36 08 | 截圖 2022-10-03 上午9 35 39 |

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/pages/dashboard-page.jsx [hunk 1]
```diff
@@ -147,13 +168,29 @@ export default function DashboardPage( { isLoading = false } ) {
...
+	};
+
 	return (
 		<>
 			{ isPageLoading && <Loading /> }
 			{ ! isPageLoading && (
 				<div className="jp-search-dashboard-page">
 					{ renderHeader() }
 					{ renderMockedSearchInterface() }
+					{ renderDonutMeterTestSection() }
```
Comment (oskosk) [inline projects/packages/search/src/dashboard/components/pages/dashboard-page.jsx]: Let's not do this. There's no point in using functions called `renderSomething` when components are actually functions that just return what's gonna be rendered. If you go with this approach this function is not visible for the React Dev Tools as a component. 

Go for a name like `const DonutMeterTestSection = ()...` and use it like this `<DonutMeterTestSection />`

In this particular example I understand there were `renderHeader` and `renderMockedSearchInterface` not introduced in the PR, but let's just do it properly with `DonutMeterTestSection`

Comment (a8ck3n) [reply projects/packages/search/src/dashboard/components/pages/dashboard-page.jsx:193]: Thanks for the feedback @oskosk! Much appreciated!

In this case, I'm building out `DonutMeterWithLabels` as the actual component but I understand what you're getting at here. Maybe moving this code block into a "layout" component is the best option here? I'll give that a try as I continue to build this out.

Comment (a8ck3n) [reply projects/packages/search/src/dashboard/components/pages/dashboard-page.jsx]: Thanks for the feedback @oskosk! Much appreciated!

In this case, I'm building out DonutMeterWithLabels as the actual component but I understand what you're getting at here. Maybe moving this code block into a "layout" component is the best option here? I'll give that a try as I continue to build this out.

**Comment by oskosk** [comment]
Pushed a rebase in [171f874](https://github.com/Automattic/jetpack/pull/26344/commits/171f874d1684404d87117c3cf0ea9e5976581eff) and [7c04d95](https://github.com/Automattic/jetpack/pull/26344/commits/7c04d95c24cdf56d07a90eacffbfca89f224c2ed)

**Comment by kangzj** [comment]
I'll shepherd this further after I mock the required APIs and if no one gets on it first. 🐑

**Review by kangzj — APPROVED** [review]
Looks good to me 👍 

I tweaked the number formatting a bit because some countries use dot as thousand separators rather than commas.

Another thing I noticed is that the donut meters don't change colors with the usage, like if the usage is above 100%, then the meter should be red.

Thanks for shepherding this @dognose24. Great job! 👍

**Comment by dognose24** [comment]
> I tweaked the number formatting a bit because some countries use dot as thousand separators rather than commas.

Thanks for the more general solution!

> Another thing I noticed is that the donut meters don't change colors with the usage, like if the usage is above 100%, then the meter should be red.

Sure thing! There should be another PR to address donut colors and event callbacks.

**Pull request merged**